flash/text/classification/model.py

```diff
@@ -88,12 +82,45 @@ def __init__(
         )
         self.enable_ort = enable_ort
         self.model = self.backbones.get(backbone)(num_labels=num_classes)
+        self.pretrained = pretrained
+
+        if self.pretrained:
+            if vocab_size:
+                print("`pretrained=True`, ignoring `vocab_size` argument.")
+            self.vocab_size = self.model.config.vocab_size
+
+        else:
+            if vocab_size:
+                self.vocab_size = vocab_size
+                print(f"Re-initialize word embeddings layer with `vocab_size={self.vocab_size}`")
+            else:
+                self.vocab_size = self.model.config.vocab_size
+                print(f"Re-initialize word embeddings layer with the original `vocab_size={self.vocab_size}`")
+            self._init_embeddings()
+
         self.save_hyperparameters()
 
     @property
     def backbone(self):
         return self.model.base_model
 
+    def _init_embeddings(self):
+        num_embeddings = self.model.config.vocab_size
+        initializer_range = self.model.config.initializer_range
+
+        for name, module in self.model.named_modules():
+            # find the word embedding layer
+            if isinstance(module, torch.nn.Embedding) and module.num_embeddings == num_embeddings:
```

Comment:
What if the instance of the Embedding isn't torch.nn.Embedding (shared embeddings as an example)?

Reply:
I agree the current implementation in hacky.

As a reference, this is the structure of the embedding of BERT
```python
BertModel(
    (embeddings): BertEmbeddings(
    (word_embeddings): Embedding(30522, 128, padding_idx=0)
    (position_embeddings): Embedding(512, 128)
    (token_type_embeddings): Embedding(2, 128)
    (LayerNorm): LayerNorm((128,), eps=1e-12, elementwise_affine=True)
    (dropout): Dropout(p=0.1, inplace=False)
)
```

In order to identify the correct piece to substitute, what do you think if I use the original `vocab_size` of the model as a filtering criterion and the "embed" regex (this would include `nn.Embedding` and `nn.EmbeddingBag` modules)?

```python
for name, param in model.named_parameters():
    if "embed" in name.lower() and param.shape[0] == vocab_size:
        ... do things...
```